src/middlewared/middlewared/plugins/ups.py

```diff
@@ -101,12 +101,11 @@ async def validate_data(self, data, schema):
                     'Use alphanumeric characters, ".", "-" and "_"'
                 )
 
-        for field in [field for field in ['monpwd', 'monuser'] if data.get(field)]:
-            if re.search(r'[ #]', data[field], re.I):
-                verrors.add(
-                    f'{schema}.{field}',
-                    'Spaces or number signs are not allowed'
-                )
+        for field in ['monpwd', 'monuser']:
+            if not data.get(field):
+                verrors.add(f'{schema}.{field}', 'This field is required.')
+            elif re.search(r'[ #]', data[field], re.I):
+                verrors.add(f'{schema}.{field}', 'Spaces or number signs are not allowed')
```

Comment:
We need to be consistent because one ends with a dot and the other error doesnt.

Also does this need backport to 12?

Reply:
Yes it does, making another PR for that. Thank you for catching it